### ISSUE TITLE
[Anon] Same cat IDs found

### ISSUE THREAD

**Issue by BSDataAnon** [BSData/wh40k-7th-edition#246]
Error appears when loading 7th edition 40k v3:

Errors preparing data:

> More than one catalogue found with the same id.

I have tried deleting rosters and data files and redownloading, did not fix the issue.

**Comment by Kangodo** [comment]
Are you sure you deleted everything? Because that error happens when you leave old files in your folders.

**Comment by Jonskichov** [comment]
I've seen this come up a lot lately. I'll check the data import on my end (it shouldn't end up with multiple files with the same ID), but is there this problem in the repo? (I suspect the Wolves file since the old one is still in there, but haven't checked the IDs)

**Comment by Jonskichov** [comment]
...it's not the wolves one.

A thought, are there more than one game systems (in different repos) with the same ID?

**Comment by amis92** [comment]
Well I have checked in Notepad++, all 46 catalogues have different IDs in wh40k.

I can only check our BSData repo for duplicate gst-IDs.

**Comment by amis92** [comment]
These are IDs of all our game systems on BSData. All unique:

```
fc079761-2937-49e8-bf32-0b25e0ed8da6
e1ebd931-a209-3ce4-87b4-d9918d25530b
be4eb679-97dc-4876-b582-19ff87fae0fd
ca571888-56a9-c58e-ddaf-54f4713538bc
b978a3ef-974a-4354-b82d-7965f519d3df
e37e48f3-bb87-e20a-cf08-355a001c0e6a
17f81b5a-0320-dfae-b485-6d856bff4ae2
dea5da1e-8b51-b068-a829-a7b0a88b17d4
417a1fa3-3f8e-7789-d5ec-a4ebd35323cd
dd1c28f5-7a1e-e616-4caa-87ff07e7d4f1
003c16ad-cf7a-5f7f-924c-68fcbae35907
9481a749-7900-614b-1695-bdc2899069c1
9b21e8c8-31b6-82f0-9805-ad62649f1f3d
22ac4872-6f49-f3a0-8c06-60e9ad204df5
01e31045-05fb-42cc-9803-cd07615b4148
c32ab30b-7a2f-4e2a-a46f-e5c249e706cd
de5d6c5f-ae7b-4dd1-841e-5f1193fb5176
bfef4c13-8961-4056-a7ab-30a35cfaf51c
d18e75d8-f27e-b243-cb33-25f023403189
5ff82c92-8faf-fee2-9d9e-531d4753b8c8
6f905f96-08d9-4b54-8ed7-cd0a2a747bf4
```

**Comment by katfude** [comment]
I had this occur to me today, but that was because I was dumb and had 2 different sets of 40k folders in my battlescribe directory. I'm willing to bet a lot of money this is what occurred to the anonymous user. I suggest this issue be closed.

**Comment by amis92** [comment]
Ok @BSData/data-admins @BSData/data-authors @Jonskichov (I think you all should know for future)

I know what happened. When new Space Wolves codex came out, we created new file with new id, renamed the old one and released it.

On BattleScribe user's end updates are resolved on the basis of index.bsi file it downloads from repo. It then compares revision numbers of files it has in memory already, with those listed in index file.

There appeared to be such a situtation:

| local memory | in index |
| --- | --- |
| Space Wolves - Codex.cat (old id) | Space Wolves - Codex (2009).cat (old id) |
|  | Space Wolves - Codex.cat (new id) |

What happened (local memory):

| before update | after update |
| --- | --- |
| Space Wolves - Codex.cat (old id) | Space Wolves - Codex.cat (old id) |
|  | Space Wolves - Codex _a-very-long-digit-string-for-time_.cat (new id) |

(that's an effect a user reported in #198 and #199 and khambatta suggested)

Then there came duplicate.

**To sum up**: BattleScribe doesn't update filenames as we do here in repository. It kinda sucks, but we can't do anything. We cannot therefore name files without including a year, because in future there will be that same problem. **From this moment onwards, any new files should be named according to new [guidelines](https://github.com/BSData/bsdata/wiki/Catalogue-Naming-Convention)**.

**Comment by ghost** [comment]
If I hadn't made a new ID for the new SW codex, this wouldn't've happened. _Insert sheepish look here._

Did that so people could keep the old file, but is there really a need?

**Comment by Kangodo** [comment]
Interesting, because I haven't heard about issues with the GK-Codex and I did the same with those.
But yes, I think it's needed.
We are an awesome team and have very quick releases, but I don't expect everyone to buy the Codex within three days.

**Comment by amis92** [comment]
@Kangodo no, you haven't renamed the old file, and didn't change its ID. So in reality you only created new catalogue with new id, which doesn't cause problems.

@khambatta I think it's ok to have old files, we just need not to change their names. See last comments in #201 .
